lasagne/tests/test_init.py

```diff
@@ -15,55 +15,137 @@ def test_normal():
+
+
+def test_glorot_normal_gain():
+    from lasagne.init import GlorotNormal
+
+    sample = GlorotNormal(gain=10.0).sample((100, 100))
+    assert -0.1 < sample.mean() < 0.1
+    assert 0.9 < sample.std() < 1.1
 
-    sample = Uniform().sample((150, 450))
+
+def test_glorot_normal_c01b():
+    from lasagne.init import GlorotNormal_c01b
+
+    sample = GlorotNormal_c01b().sample((25, 2, 2, 25))
+    assert -0.01 < sample.mean() < 0.01
+    assert 0.09 < sample.std() < 0.11
+
+
+def test_glorot_normal_c01b_4d_only():
+    from lasagne.init import GlorotNormal_c01b
+
+    with pytest.raises(RuntimeError):
+        GlorotNormal_c01b().sample((100,))
+
+    with pytest.raises(RuntimeError):
+        GlorotNormal_c01b().sample((100, 100))
+
+    with pytest.raises(RuntimeError):
+        GlorotNormal_c01b().sample((100, 100, 100))
+
+
+def test_glorot_uniform():
+    from lasagne.init import GlorotUniform
+
+    sample = GlorotUniform().sample((150, 450))
     assert -0.11 < sample.min() < -0.09
     assert 0.09 < sample.max() < 0.11
 
 
-def test_uniform_glorot_receptive_field():
-    from lasagne.init import Uniform
+def test_glorot_uniform_receptive_field():
+    from lasagne.init import GlorotUniform
 
-    sample = Uniform().sample((150, 150, 2))
+    sample = GlorotUniform().sample((150, 150, 2))
     assert -0.11 < sample.min() < -0.09
     assert 0.09 < sample.max() < 0.11
 
 
-def test_uniform_range_as_number():
-    from lasagne.init import Uniform
+def test_glorot_uniform_gain():
+    from lasagne.init import GlorotUniform
 
-    sample = Uniform(1.0).sample((300, 400))
-    assert sample.shape == (300, 400)
+    sample = GlorotUniform(gain=10.0).sample((150, 450))
     assert -1.1 < sample.min() < -0.9
     assert 0.9 < sample.max() < 1.1
 
 
-def test_uniform_range_as_range():
-    from lasagne.init import Uniform
+def test_glorot_uniform_c01b():
+    from lasagne.init import GlorotUniform_c01b
 
-    sample = Uniform((0.0, 1.0)).sample((300, 400))
-    assert sample.shape == (300, 400)
-    assert -0.1 < sample.min() < 0.1
-    assert 0.9 < sample.max() < 1.1
+    sample = GlorotUniform_c01b().sample((75, 2, 2, 75))
+    assert -0.11 < sample.min() < -0.09
+    assert 0.09 < sample.max() < 0.11
+
+
+def test_glorot_uniform_c01b_4d_only():
+    from lasagne.init import GlorotUniform_c01b
+
+    with pytest.raises(RuntimeError):
+        GlorotUniform_c01b().sample((100,))
+
+    with pytest.raises(RuntimeError):
+        GlorotUniform_c01b().sample((100, 100))
+
+    with pytest.raises(RuntimeError):
+        GlorotUniform_c01b().sample((100, 100, 100))
+
+
+def test_constant():
+    from lasagne.init import Constant
+
+    sample = Constant(1.0).sample((10, 20))
+    assert (sample == 1.0).all()
+
+
+def test_sparse():
+    from lasagne.init import Sparse
+
+    sample = Sparse(sparsity=0.5).sample((10, 20))
+    assert (sample == 0.0).sum() == (sample != 0.0).sum()
+    assert (sample == 0.0).sum() == (10 * 20) / 2
```

Comment:
Hmm, what if someone changes `sparsity` to behave like `1 - sparsity`? That would go unnoticed. I'd prefer:
```python
sample = Sparse(sparsity=0.1).sample((10, 20))
assert (sample != 0.0).sum() == (10 * 20) * 0.1
```
(There's no need for a second assert unless you want to ensure `sample` has 10*20 items.)